src/main/webapp/app/markdown-editor/markdown-editor.component.ts

```diff
@@ -0,0 +1,255 @@
+     *           5b. All the text between two identifiers is mapped to the first identifier
+     *       6. Reduce the copy by the length of the command
+     *       7. Call the parseLineForDomainCommand for command and save it into content
+     *       8. Emit the content to parent component to assign the values of the array to the right attributes
+     */
+    parse(): void {
+        /** check if domainCommands are passed on from the parent component */
+        if (this.domainCommands == null || this.domainCommands.length === 0) {
+            /** if no domainCommands contained emit the markdown text converted to html to parent component to display */
+                this.previewTextAsHtml = this.artemisMarkdown.htmlForMarkdown(this.markdown);
+                this.html.emit(this.previewTextAsHtml);
+            return;
+        } else {
+            /** create array with domain command identifier */
+            const possibleCommandIdentifier = this.domainCommands.map(command => command.getOpeningIdentifier());
+            /** create empty array which
+             * will contain the splitted textline with the corresponding domainCommandIdentifier which
+             * will be emitted to the parent component */
+            const commandTextsMappedToCommandIdentifiers = [];
+            /** create a copy of the markdown text to loop trough it and find the domainCommandIdentifier */
+            let copy = this.markdown.slice(0);
+
+            /** create array with the identifiers to use for RegEx by deleting the [] of the domainCommandIdentifiers */
+            const tagNames = possibleCommandIdentifier.map(tag => tag.replace('[', '').replace(']', '')).join('|');
+
+            /** create a new regex expression which searches for the domainCommands identifiers
+             * (?=   If a command is found, add the command identifier to the result of the split
+             * \\[  look for the character '[' to determine the beginning of the command identifier
+             * (${tagNames}) look if after the '[' one of the element of tagNames is contained
+             * \\] look for the character ']' to determine the end of the command identifier
+             * )  close the bracket
+             *  g: search in the whole string
+             *  m: match the regex over multiple lines*/
+            const regex = new RegExp(`(?=\\[(${tagNames})\\])`, 'gm');
+
+            /** iterating loop as long as the copy of the markdown text exists and split the copy as soon as a domainCommand identifier is found */
+            while (copy.length) {
+                /** As soon as an identifier is with regEx the copy of the markdown text is split and saved into {array} command
+                 *  split: saves its values into an {array}
+                 *  limit 1: indicated that as soon as an identifier is found copy is split */
+                const [command] = copy.split(regex, 1);
```

Comment:
I don't like the term `command` here, I still don't understand what the array contains. Is it the text between/before/after the commands? I think we need to include the word Text within the variable and I suggest to use a rather long variable name that immediately tells the developer what the variable *exactly* contains